nntrainer/src/tensor.cpp

```diff
@@ -61,50 +61,36 @@ static auto rng = [] {
   return rng;
 }();
 
-Tensor::Tensor(const TensorDim d) {
-  dim = d;
-  this->data = std::vector<float>(dim.getDataLen());
-  _is_contiguous = true;
-  setZero();
-}
-
-Tensor::Tensor(int height, int width) {
-  dim.height(height);
-  dim.width(width);
-  this->data = std::vector<float>(dim.getDataLen());
-  _is_contiguous = true;
-  setZero();
+Tensor &Tensor::operator=(const Tensor &rhs) {
+  Tensor tmp(rhs);
+  std::swap(*this, tmp);
+  return *this;
 }
 
-Tensor::Tensor(int channel, int height, int width) {
-  dim.height(height);
-  dim.width(width);
-  dim.batch(channel);
-  this->data = std::vector<float>(dim.getDataLen());
-  _is_contiguous = true;
-  setZero();
-}
+Tensor &Tensor::operator=(Tensor &&rhs) noexcept {
+  std::swap(dim, rhs.dim);
```

Comment:
this works too but its recommended to create a internal swap function for this.

Reply:
AFAIK, from c++11 it is find to use `std::swap` because it does not use copy constructors if possible which wasn't the case before c++11. 
However, I think I should add move ctor in `TensorDim` later on. 
Thanks for pointing out and please tell me if I got something wrong!

cf) http://www.cplusplus.com/reference/algorithm/swap/

Reply:
yes, you are right.
what I meant what create a function `void Tensor::swap(Tensor &lhs, Tensor &rhs)`.

Reply:
Added :) Thanks.